src/h5web/App.test.tsx

```diff
@@ -1,23 +1,134 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import {
+  render,
+  RenderResult,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
 import App from './App';
 import MockProvider from './providers/mock/MockProvider';
 
 const DOMAIN = 'bsa_002_000-integrate-sub';
 
-test('renders with root group selected', async () => {
-  const { findByText, findByRole } = render(
+function renderApp(): RenderResult {
+  return render(
     <MockProvider domain={DOMAIN}>
       <App />
     </MockProvider>
   );
+}
+
+test('renders with root group selected', async () => {
+  const { findByText, getByRole } = renderApp();
 
   const noVisText = await findByText('Nothing to visualize');
-  expect(noVisText).toBeInTheDocument();
+  expect(noVisText).toBeVisible();
 
-  const title = await findByRole('heading', { name: DOMAIN });
-  expect(title).toBeInTheDocument();
+  const title = getByRole('heading', { name: DOMAIN });
+  expect(title).toBeVisible();
 
-  const domainBtn = await findByRole('treeitem', { name: DOMAIN });
+  const domainBtn = getByRole('treeitem', { name: DOMAIN });
+  expect(domainBtn).toBeVisible();
   expect(domainBtn).toHaveAttribute('aria-selected', 'true');
 });
+
+test('switch between "display" and "inspect" modes', async () => {
+  const {
+    findByRole,
+    findByText,
+    getByRole,
+    queryByRole,
+    queryByText,
+  } = renderApp();
+
+  const inspectBtn = getByRole('tab', { name: 'Inspect' });
+  const displayBtn = getByRole('tab', { name: 'Display' });
+
+  // Switch to "inspect" mode
+  fireEvent.click(inspectBtn);
+
+  const noVisText1 = queryByText('Nothing to visualize');
+  expect(noVisText1).not.toBeInTheDocument();
```

Comment:
`queryByText` doesn't throw if the element isn't in the DOM, which is useful to expect this very fact.

Reply:
if the element disappears asynchronously, see my comment further down